src/foam/nanos/http/LiveScriptBundler.java

```diff
@@ -31,6 +32,7 @@
 
   // Caching
   protected String javascriptBuffer_;
+  protected ReentrantLock requestBlockingLock_;
```

Comment:
Is this needed? Is the doRebuildJavascript() method reentrant? Couldn't you just make that method synchronized and then to a synchronized (this) {} catch(InterruptedException e) {} in execute()?

Reply:
Do synchronized methods and synchronized blocks interact that way, or would I need to add two synchronized blocks instead?

Reply:
After doing some reading, I think that would still allow a request to go through during building. The reason is that it would behave equivalently to having two separate reentrant locks (one to each synchronized block)